oxidation-of-metals.csl

```diff
@@ -0,0 +1,151 @@
+      <name sort-separator=" " delimiter=", " initialize-with="" delimiter-precedes-last="always" name-as-sort-order="all"/>
+      <label form="short" prefix=", "/>
+      <et-al font-style="normal"/>
+    </names>
+  </macro>
+  <macro name="access">
+    <choose>
+      <if variable="volume"/>
+      <else-if variable="DOI">
+        <text variable="DOI" prefix="doi:"/>
+      </else-if>
+      <else-if type="webpage" variable="URL" match="all">
+        <text term="available at" text-case="capitalize-first" suffix=": "/>
+        <text variable="URL" suffix=". "/>
+        <group prefix="" suffix="" delimiter=" ">
+          <text term="accessed" text-case="capitalize-first"/>
+          <date variable="accessed">
+            <date-part name="day" suffix=" " form="ordinal"/>
+            <date-part name="month" suffix=" "/>
+            <date-part name="year"/>
+          </date>
+        </group>
+      </else-if>
+    </choose>
+  </macro>
+  <macro name="issuance">
+    <choose>
+      <if type="bill book graphic legal_case legislation motion_picture report song thesis paper-conference" match="any">
+        <group>
+          <text variable="publisher-place" form="long" suffix=": "/>
+          <text variable="publisher" form="long" suffix="; "/>
+          <date variable="issued">
+            <date-part name="year" suffix="."/>
+          </date>
+        </group>
+      </if>
+      
+      <else-if type="chapter" match="any">
+        <group>
+          <text variable="publisher-place" form="long" suffix=": "/>
+          <text variable="publisher" form="long" suffix="; "/>
+          <date variable="issued">
+            <date-part name="year" suffix=":"/>
+          </date>
+        </group>
+      </else-if>
+      
+      <else>
+        <date prefix="" suffix=";" variable="issued">
+          <date-part name="year"/>
+        </date>
+      </else>
+    </choose>
+  </macro>
+  <macro name="container-title">
+    <choose>
+      <if type="article-journal">
+        <text variable="container-title" font-style="italic" form="short" suffix=". "/>
+      </if>      
+      <else>
+        <text variable="container-title" font-style="italic" suffix=". "/>
+      </else>
+    </choose>
+  </macro>
+  <macro name="editor">
+    <choose>
+      <if type="chapter paper-conference" match="any">
+        <names variable="editor" prefix="" suffix="">
+          <name delimiter-precedes-last="never" initialize-with="" name-as-sort-order="all" sort-separator=" "/>
+          <label form="short" suffix=" " prefix=", "/>
+        </names>
+      </if>
+    </choose>
+  </macro>
+  <citation collapse="citation-number">
+    <sort>
+      <key variable="citation-number"/>
+    </sort>
+    <layout prefix="[" suffix="]" delimiter=",">
+      <text variable="citation-number"/>
+    </layout>
+  </citation>
+  <bibliography et-al-min="7" et-al-use-first="3" second-field-align="flush" entry-spacing="0" line-spacing="2">
+    <layout>
```

Comment:
into layout, add 'suffix="."' so there is a dot at the end of each bibliograpy entry.